samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/MultiLingualBot.java

```diff
@@ -1,2 +1,173 @@
+import com.microsoft.bot.builder.UserState;
+import com.microsoft.bot.builder.StatePropertyAccessor;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.schema.ChannelAccount;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.CardAction;
+import com.microsoft.bot.schema.ActionTypes;
+import com.microsoft.bot.schema.SuggestedActions;
+import com.microsoft.bot.schema.Attachment;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This bot demonstrates how to use Microsoft Translator.
+ * More information can be found
+ * here https://docs.microsoft.com/en-us/azure/cognitive-services/translator/translator-info-overview.
+ */
+public class MultiLingualBot extends ActivityHandler {
+    private static final StringBuilder WELCOME_TEXT =
+        new StringBuilder("This bot will introduce you to translation middleware. ")
+        .append("Say 'hi' to get started.");
+    private static final String ENGLISH_ENGLISH = "en";
+    private static final String ENGLISH_SPANISH = "es";
+    private static final String SPANISH_ENGLISH = "in";
+    private static final String SPANISH_SPANISH = "it";
+
+    private UserState userState;
+    private StatePropertyAccessor<String> languagePreference;
+
+    /**
+     * Creates a Multilingual bot.
+     * @param withUserState User state object.
+     */
+    public MultiLingualBot(UserState withUserState) {
+        if (withUserState != null) {
+            userState = withUserState;
+        } else {
+            throw new NullPointerException("userState");
+        }
+        languagePreference = userState.createProperty("LanguagePreference");
+    }
+
+    /**
+     * This method is executed when a user is joining to the conversation.
+     * @param membersAdded A list of all the members added to the conversation,
+     *                     as described by the conversation update activity.
+     * @param turnContext The context object for this turn.
+     * @return A task that represents the work queued to execute.
+     */
+    @Override
+    protected CompletableFuture<Void> onMembersAdded(List<ChannelAccount> membersAdded,
+                                                     TurnContext turnContext) {
+        return this.sendWelcomeMessage(turnContext);
+    }
+
+    /**
+     * This method is executed when the turnContext receives a message activity.
+     * @param turnContext The context object for this turn.
+     * @return A task that represents the work queued to execute.
+     */
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
+        if (this.isLanguageChangeRequested(turnContext.getActivity().getText())) {
```

Comment:
```suggestion
        if (MultiLingualBot.isLanguageChangeRequested(turnContext.getActivity().getText())) {
```

Because the method is `static`